library/src/main/kotlin/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -72,7 +74,12 @@ public class ChuckerInterceptor private constructor(
     override fun intercept(chain: Interceptor.Chain): Response {
         val transaction = HttpTransaction()
         val request = chain.request()
-        val shouldProcessTheRequest = !skipPaths.any { it == request.url.encodedPath }
+        val path = request.url.encodedPath
+        val shouldSkipPath = skipPaths.contains(path) || skipPathsRegex.any { it.matches(path) }
+        // Skip evaluation of domain if the path is already skipped
+        val shouldSkipDomain = shouldSkipPath || skipDomainRegex.any { it.matches(request.url.host) }
+        val shouldProcessTheRequest = !(shouldSkipPath || shouldSkipDomain)
```

Comment:
Can you export this to an `internal` function and unit test it?

Reply:
This has been unit tested, should we test explicitly?
Ref: `chucker skips requests when skip domains are provided and matches`, `chucker should not skip requests when skip domains are provided and does not match`

Reply:
Yup the other test is probably sufficient.
This was more of a nit.

I thought we could have a:

```
internal fun shouldProcessRequest(input: Request, skipPaths, skipPathsRegex, skipDomains, skipDomainsRegex): Boolean
```

but your tests already cover them